[14.0][MIG] account_invoice_download_ovh

alexis-via opened the pull request:
This migration also contains a fix on the download of the OVH PDF invoice (I'll backport it to v10 soon)
Requires #395 
Fully tested with my OVH account.

alexis-via committed:
Rename module ovh_supplier_invoice to account_invoice_download_ovh

alexis-via committed:
OVH: use the account_invoice_download framework

Port to OVH API (instead of SoAPI)

alexis-via committed:
Improve log handling in ovh backend

Code cleanup

oca-travis committed:
[UPD] Update account_invoice_download_ovh.pot

OCA-git-bot committed:
[ADD] icon.png

njeudy committed:
[MIG] migrate account_invoice_download and account_invoice_download_ovh to V12.0

njeudy committed:
[FIX] product.product_uom_unit to uom.product_uom_unit

oca-travis committed:
[UPD] Update account_invoice_download_ovh.pot

ivantodorovich committed:
[FIX] Fix _description in model

oca-travis committed:
[UPD] Update account_invoice_download_ovh.pot

OCA-git-bot committed:
account_invoice_download_ovh 12.0.1.0.1

alexis-via committed:
[MIG] account_invoice_download_ovh from v12 to v14

alexis-via committed:
account_invoice_download_ovh: black, isort, etc.

davejames commented on account_invoice_download_ovh/models/account_invoice_download_config.py:
```diff
@@ -0,0 +1,229 @@
+                "amount_untaxed": res_inv["priceWithoutTax"].get("value"),
+                "amount_total": res_inv["priceWithTax"].get("value"),
+            }
+            self.ovh_invoice_attach_pdf(parsed_inv, res_inv["url"])
```
`res_inv["url"]` will give you an html representation of the invoice, the PDF is available at `res_inv["pdfUrl"]`

```suggestion
            self.ovh_invoice_attach_pdf(parsed_inv, res_inv["pdfUrl"])
```

https://api.ovh.com/console/#/me/bill/{billId}#GET

davejames reviewed (changes requested):
Other than the fact that PDFs download as HTML this works well for me (fix suggested). Tested with my OVH account on ovh-ca endpoint

alexis-via committed:
ovh: use pdfUrl instead of url

res_inv['pdfUrl'] always return a PDF whereas res_inv['url'] sometimes
returns a PDF and sometimes return an HTML (I've seen both, maybe the
difference is made on the HTTP client... I don't know exactly).

davejames reviewed (approved):
functional test, works well for me

alexis-via commented:
/ocabot merge nobump
